wallet/wallet-cli-lib/src/repl/interactive/mod.rs

```diff
@@ -226,3 +231,93 @@ fn handle_response<N: NodeInterface>(
     }
     None
 }
+
+fn paginate_output(
+    header: String,
+    body: String,
+    line_editor: &mut Reedline,
+    console: &mut impl ConsoleOutput,
+) {
+    let mut current_index = 0;
+
+    let (cols, rows) = crossterm::terminal::size().unwrap_or((80, 24));
+    let cols = cols as usize;
+    let page_rows = (rows - 4) as usize; // make room for the header and prompt
+
+    loop {
+        line_editor.clear_screen().expect("Should not fail normally");
+
+        let limit = compute_visible_text_limit(
+            body.get(current_index..).expect("safe point").lines(),
+            cols,
+            page_rows,
+        );
+        let end_index = std::cmp::min(current_index + limit, body.len());
+
+        console.print_line(&header);
+        console.print_line(body.get(current_index..end_index).expect("safe point"));
+
+        let commands = match (current_index, end_index) {
+            (0, end) if end == body.len() => "Press 'q' to quit",
+            (0, _) => "Press 'j' for down, 'q' to quit",
+            (_, end) if end == body.len() => "Press 'k' for previous, 'q' to quit",
+            (_, _) => "Press 'j' for next, 'k' for previous, 'q' to quit",
+        };
+        console.print_line(commands);
+
+        // Wait for user input.
+        crossterm::terminal::enable_raw_mode().expect("Should not fail normally");
+        let event = crossterm::event::read().expect("Should not fail normally");
+        crossterm::terminal::disable_raw_mode().expect("Should not fail normally");
+
+        if let crossterm::event::Event::Key(key_event) = event {
```

Comment:
It would be nice to handle the `Resize` event as well (a TODO is fine for me)

Reply:
fixed